db/sstables-format-selector.cc

```diff
@@ -75,7 +74,6 @@ future<> sstables_format_selector::do_maybe_select_format(sstables::sstable_vers
 future<> sstables_format_selector::start() {
     assert(this_shard_id() == 0);
     return read_sstables_format().then([this] {
-        _features.local().cluster_supports_mc_sstable().when_enabled(_mc_feature_listener);
```

Comment:
I think this might be unsafe. UNBOUNDED_RANGE_TOMBSTONE is being enabled by this listener. I believe that you you need to enable UNBOUNDED_RANGE_TOMBSTONE by default if you're removing this listener.

Reply:
Probably just removing this line will do the job:
https://github.com/scylladb/scylla/blob/10364fca6ee41aa1697edfc80e67d08ce6c26137/gms/feature_service.cc#L100

Reply:
Another question is whether we can remove _mc_feature_listener althogether?

Reply:
It doesn't really get enabled by this listener - in this series, `_selected_format` defaults to `mc`, so this feature is masked and then unmasked unconditionally. This masking-and-then-unmasking makes little sense and it should be purged, but it should nonetheless work fine.

And I did remove _mc_feature_listener altogether in this series, because the `_selected_format` now defaults to mc. Whether it makes sense or not is a separate issue and I should at least mention it in the commit message though.

Reply:
My main source of confusion was that I assumed that `MC_SSTABLE_FORMAT` feature was only used to announce that a node **can** support mc sstables, while in reality it's also used to actually decide if mc format is used or not. That complicates the whole business with deprecating features.